dapp/src/hooks/cardano/useGetLocksCardano.ts

```diff
@@ -0,0 +1,107 @@
+"use client";
+import { LockInfoCardano } from "../../utils/cardano/types";
+import FunctionKey from "../../utils/functionKey";
+import { useQuery } from "@tanstack/react-query";
+import axios from "axios";
+import { useDappStore } from "../../store";
+import { Asset } from "../../utils/cardano/asset";
+import { Token } from "../../utils/cardano/token";
+import {
+  cardanoApiMaxSlot,
+  cardanoApiMinSlot,
+} from "../../utils/cardano/constants";
+import env from "../../utils/configs/env";
+import { ProjectedNftRangeResponse } from "@dcspark/carp-client/shared/models/ProjectedNftRange";
+
+const fetchLocks = async (paymentKeyHash: string) => {
+  try {
+    const request = await axios.post(
+      `${env.REACT_APP_CARDANO_API_URL_BASE}/projected-nft/range`,
+      { range: { minSlot: cardanoApiMinSlot, maxSlot: cardanoApiMaxSlot } },
+    );
+    const responseData: ProjectedNftRangeResponse = request.data;
+    const locksMap: Record<string, LockInfoCardano> = {};
+    responseData.forEach((dat) => {
+      if (dat.status === "Invalid") return;
```

Comment:
Reminder: we'll want to replace these magic strings later when we have the projected NFT contract deployed to NPM